pkg/model/awsmodel/iam.go

```diff
@@ -333,10 +334,8 @@ func (b *IAMModelBuilder) buildIAMTasks(role iam.Subject, iamName string, c *fi.
 			}
 
 			c.AddTask(t)
-		}
 
-		// Generate additional policies if needed, and attach to existing role
-		if !shared {
+			// Generate additional policies if needed, and attach to existing role
```

Comment:
Having this as a block would make it more readable, but your choice.
```suggestion
			// Generate additional policies if needed, and attach to existing role
			{
```